api-scripts/construct_mappreview.py

```diff
@@ -0,0 +1,367 @@
+        vector_info = json.loads(vector_metadata['json'])['tilestats']
+        if vector_info['layerCount'] > 1:
+            LOGGER.warning(
+                f"Vector has more than 1 layer; only using the first: {url}")
+
+        layer = vector_info['layers'][0]
+
+        return {
+            "name": vector_resource['name'],
+            "type": "vector",
+            "url": url,
+            "bounds": bounds,
+            "vector_type": layer['geometry'],
+            "feature_count": layer['count'],
+        }
+    except Exception:
+        LOGGER.exception(f"Could not load MVT {url}")
+        return None
+
+
+def get_geojson_layer_metadata(vector_resource: dict) -> dict:
+    """Get metadata needed to display a raster layer in the map"""
+    vector_type = None
+    feature_count = None
+    bounds = [-180, -90, 180, 90]
+
+    # Confirm data exists at url, get some information about it
+    url = vector_resource['url']
+    try:
+        with urllib.request.urlopen(url) as req:
+            data = json.load(req)
+
+            features = data['features']
+            feature_count = len(features)
+            vector_type = features[0]['geometry']['type']
+    except Exception as e:
+        LOGGER.exception(f'Failed to access vector dataset {url}')
+        return None
+
+    # Get bounds from metadata
+    metadata_url = vector_resource['metadata_url']
+    try:
+        with urllib.request.urlopen(metadata_url) as req:
+            yaml_data = yaml.safe_load(req.read())
+            bounding_box = yaml_data['spatial']['bounding_box']
+            bounds = [
+                bounding_box['xmin'],
+                bounding_box['ymin'],
+                bounding_box['xmax'],
+                bounding_box['ymax'],
+            ]
+    except Exception as e:
+        LOGGER.exception(f'Failed to access vector metadata {metadata_url}')
+        return None
+
+    return {
+        'name': vector_resource['name'],
+        'type': 'vector',
+        'url': url,
+        'bounds': bounds,
+        'vector_type': vector_type,
+        'feature_count': feature_count,
+    }
+
+
+def get_vector_layers_metadata(vector_resources):
+    vector_layers = []
+    for vector_layer in vector_resources:
+        if not vector_layer:
+            continue
+        vector_layers.append(get_vector_layer_metadata(vector_layer))
+    return filter(None, vector_layers)
+
+
+def get_mappreview_metadata(resources, source_files, mappreview_sources=[]):
+    """Get metadata needed to display all resources in the map"""
+    raster_resources = [r for r in resources if r['format'] == 'GeoTIFF']
+    vector_resources = [r for r in resources if r['format'] == 'Shapefile']
+    layers = []
+
+    zip_resource = next((r for r in resources if r['format'] == 'ZIP'), None)
+
+    if zip_resource and source_files:
+        # Look at zip sources for spatial resources and add
+        for shp_source in [s for s in source_files if s.endswith('shp')]:
```

Comment:
```suggestion
        for shp_source in [s for s in source_files if s.endswith('.shp')]:
```

In case there is maybe a directory that ends in "shp"?